api/server/routecache/cacher.go

```diff
@@ -0,0 +1,101 @@
+package routecache
+
+import (
+	"context"
+	"errors"
+	"github.com/iron-io/functions/api/models"
+	"github.com/iron-io/runner/common"
+	"math"
+	"sync"
+)
+
+type Cacher interface {
+	PrimeCache(ctx context.Context)
+	CacheGet(ctx context.Context, appname, path string) (*models.Route, bool)
+	RefreshCache(ctx context.Context, appname string, route *models.Route)
+	ResetCache(ctx context.Context, appname string, delta int)
+}
+
+type DefaultCacher struct {
+	Datastore models.Datastore
```

Comment:
Exactly, so probably it should be the Serve running a `primecache` function internally, that would call the plugged caches.